File: frontend/src/types/meeting.ts
// ── 회의록 관련 타입 (백엔드 DTO 1:1 매핑) ─────────────────────────────

/** 회의 응답 DTO */
export interface Meeting {
  id: string;
  projectId: string;
  title: string;
  meetingDate: string | null;   // OffsetDateTime → ISO string
  purpose: string | null;       // 회의 목적/안건
  notes: string | null;         // 회의록 내용
  decisions: string | null;     // 결정사항
  checkinCode: string;          // 6자리 체크인 코드
  aiSummary: string | null;     // AI 요약 (저장됨)
  notionPageId: string | null;  // Notion 동기화 URL
  notionSyncedAt: string | null; // 마지막 동기화 시각
  createdBy: string;
  createdAt: string;
}

/** 참석자 응답 DTO */
export interface Attendee {
  userId: string;
  name: string;
  email: string;
  checkedIn: boolean;
  checkedAt: string | null;     // 체크인 시각
}

/** 회의 생성 요청 */
export interface CreateMeetingPayload {
  title: string;
  meetingDate?: string;         // OffsetDateTime 형식: "2026-03-30T14:00:00+09:00"
  purpose?: string;
}

/** 회의 수정 요청 */
export interface UpdateMeetingPayload {
  title?: string;
  meetingDate?: string;
  purpose?: string;
  notes?: string;
  decisions?: string;
}

File: backend/src/main/java/com/blackbox/entity/Meeting.java
package com.blackbox.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "meetings")
@Getter
@Setter
@NoArgsConstructor
public class Meeting {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(nullable = false, updatable = false)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "project_id", nullable = false, updatable = false)
    private Project project;

    @Column(length = 255)
    private String title;

    @Column(name = "meeting_date", nullable = false)
    private OffsetDateTime meetingDate;

    @Column(columnDefinition = "TEXT")
    private String purpose;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(columnDefinition = "TEXT")
    private String decisions;

    @Column(name = "checkin_code", unique = true, length = 8)
    private String checkinCode;

    @Column(name = "ai_summary", columnDefinition = "TEXT")
    private String aiSummary;

    @Column(name = "notion_page_id", length = 255)
    private String notionPageId;

    @Column(name = "notion_synced_at")
    private OffsetDateTime notionSyncedAt;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "created_by", nullable = false, updatable = false)
    private User createdBy;

    @Column(name = "discord_notified", nullable = false)
    private boolean discordNotified = false;

    @Column(name = "created_at", nullable = false, insertable = false, updatable = false)
    private OffsetDateTime createdAt;
}
